File: algorithm-python/boj/boj-1246.py

f = open("test.in", "r")
input = f.readline

n, m = map(int, input().rstrip().split(' '))
parr = [int(input().rstrip()) for i in range(m)]

parr = sorted(parr,reverse=True)
ans_price, ans_tot = 0, 0
for i, val in enumerate(parr[:min(n, m)]):
    sump = (i+1) * val
    ans_tot, ans_price = (sump, val) if sump > ans_tot else (ans_tot, ans_price)

print(ans_price, ans_tot)

File: physical3/boj-1246.cc
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n, m;
    cin >> n >> m;
    vector<int> parr(m);
    for(int i=0;i<m;++i) {
        cin >> parr[i];
    }
    sort(parr.begin(), parr.end());
    int ans_tot = 0;
    int ans_price = 0;
    for(auto pi=parr.begin(); pi != parr.end(); ++pi) {
        int cnt = min((int)(parr.end() - pi), n);
        if(ans_tot < cnt * *pi) {
            ans_tot = cnt * *pi;
            ans_price = *pi;
        }
    }
    cout << ans_price << ' ' << ans_tot;
    return 0;
}
